scripts/terraform-lint.tests.ps1

```diff
@@ -19,10 +19,33 @@ else
       }
     }
 
-    Context "Do not contain the 'Owner' role" {
-      It "<Instance> does not have owner role" -TestCases $TfTestCases {
+    Context "Do not contain a disallowed 'Owner' role" {
+      It "<Instance> does not have a disallowed 'Owner' role" -TestCases $TfTestCases {
           Param($Instance)
-           ((Get-Content -raw $Instance) | Select-String -Pattern "role_definition_name.*=.*Owner" ) | should BeNullOrEmpty
+
+          [array]$roleExceptions = (
+            "Azure Event Hubs Data Owner"
+          )
+          
+          [bool]$badResultFound = $False 
+          
+          $result = ((Get-Content -raw $Instance) | Select-String -Pattern "role_definition_name.*=.*Owner" -AllMatches) 
+          $matchResults = $result.Matches.Value
+
+          foreach ($matchResult in $matchResults){
+            [array]$matchFound = @() 
+            foreach($roleException in $roleExceptions) {
+              $testResult = ($matchResult | Select-String -Pattern "role_definition_name.*=.\`"$roleException")
+              if (-not [string]::IsNullOrEmpty($testResult)) {
+                $matchFound += $true
+              }
+            }
+            if ($matchFound -notcontains $true) {
```

Comment:
is this needed? can you not have output below line 40?

Reply:
Yeah will need it as it uses that $matchFound += $true to flag if a role from the file its checking against matches a role defined in the $roleExceptions